addons/sms_twilio/tests/test_sms_twilio.py

```diff
@@ -0,0 +1,253 @@
+from contextlib import contextmanager
+from unittest.mock import patch
+
+from odoo.tools import mute_logger
+from odoo.tests import common, tagged
+from odoo.addons.base.tests.common import HttpCaseWithUserDemo
+from odoo.addons.sms.tests.common import SMSCase
+
+from odoo.addons.sms.models.sms_sms import SmsSms
+from odoo.addons.sms_twilio.tools.sms_api import SmsTwilioApi
+from odoo.addons.sms_twilio.controllers.controllers import SmsTwilioController
+
+
+class MockResponse:
+    def __init__(self, json_data, status_code):
+        self.json_data = json_data
+        self.status_code = status_code
+
+    @property
+    def ok(self):
+        return self.status_code in (200, 201)
+
+    def json(self):
+        return self.json_data
+
+
+class MockSmsTwilioApi(common.BaseCase):
```

Comment:
Could be great to include it in SMS test tools so that we can test whatever provider in tests. And then, add mass_mailing_sms tests using twillio to be sure statuses are effectively correctly propagated to mailing traces.